File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/c/ParabolaVertex.c
#include "java_arrays.h"
#include <math.h>
#include <stdint.h>

/*@
requires a != 0;
assigns \nothing;
allocates \result, \result->data;
frees \nothing;
ensures jdoublearray_valid_nonnull(\result) && \result->length == 2;
ensures \fresh(\result, sizeof(*\result)) && \fresh(\result->data, 2 * sizeof(double));
ensures \result->data[0] == (double)(-(real)b / (2.0 * (real)a));
ensures \result->data[1] == (double)((double)((double)(4.0 * (real)a * c) - (double)((real)b * b)) / (4.0 * (real)a));
*/
JDoubleArray parabolaVertex(int32_t a, int32_t b, int32_t c) {
    JDoubleArray vertex = jdouble_array_new(2);
    double x = -(double)b / (2.0 * (double)a);
    double four_a_c = 4.0 * (double)a;
    four_a_c = four_a_c * (double)c;
    double b_squared = (double)b * (double)b;
    double y = (four_a_c - b_squared) / (4.0 * (double)a);
    jdouble_array_set(vertex, 0, x);
    jdouble_array_set(vertex, 1, y);
    return vertex;
}

File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/java/ParabolaVertex.java

import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;

class ParabolaVertex {
    
    /*@
    public normal_behavior
      assignable \nothing;
      ensures \result != null && \fresh(\result) && \result.length == 2;
      ensures a != 0 ==> \result[0] == (-(double)b / (2.0 * a));
      ensures a != 0 ==> \result[1] == (((double)4.0 * a * c) - ((double)b * b)) / (4.0 * a);
      ensures a == 0 && b == 0 ==> Double.isNaN(\result[0]) && Double.isNaN(\result[1]);
      ensures a == 0 && b > 0 ==> \result[0] == Double.NEGATIVE_INFINITY;
      ensures a == 0 && b < 0 ==> \result[0] == Double.POSITIVE_INFINITY;
      ensures a == 0 && b != 0 ==> \result[1] == Double.NEGATIVE_INFINITY;
    @*/
    public static double[] parabolaVertex(int a, int b, int c) {
        double[] vertex = new double[2];
        vertex[0] = (-(double) b / (2.0 * a));
        vertex[1] = (((double) 4.0 * a * c) - ((double) b * b)) / (4.0 * a);
        return vertex;
    }
}

